rs/index/src/posting_list/file.rs

```diff
@@ -0,0 +1,699 @@
+
+        mmap[self.current_offset..self.current_offset + write_size].copy_from_slice(data);
+        self.current_offset += write_size;
+
+        Ok(())
+    }
+
+    fn offset_to_first_posting_list(&self) -> usize {
+        let pl_metadata_in_bytes = 2 * std::mem::size_of::<u64>();
+        self.num_clusters * pl_metadata_in_bytes
+    }
+
+    fn flush_resident_posting_lists_to_disk(&mut self) -> Result<()> {
+        if !self.resident {
+            return Err(anyhow!(
+                "Posting lists should still be in memory when flushing to disk"
+            ));
+        }
+        // Trigger the creation of a new file when we start to flush
+        self.current_offset = self.backing_file_size;
+        let usize_in_bytes = std::mem::size_of::<u64>();
+        // Extract all the data we need from self to avoid immutable borrow issue
+        let posting_lists: Vec<Vec<u64>> = std::mem::take(&mut self.resident_posting_lists);
+        // First write the metadata
+        let mut buffer: Vec<u8> = vec![];
+        for posting_list in &posting_lists {
+            if self.current_offset == self.backing_file_size {
+                self.new_backing_file()?;
+            }
+            buffer.extend_from_slice(posting_list.len().to_le_bytes().as_ref());
+            buffer.extend_from_slice(self.offset_to_current_posting_list.to_le_bytes().as_ref());
+            self.write_to_current_mmap(&buffer)?;
+            self.offset_to_current_posting_list += (posting_list.len() * usize_in_bytes) as u64;
+            buffer.clear();
+        }
+        // Now write the posting lists
+        self.current_offset = self.offset_to_first_posting_list();
+        for posting_list in &posting_lists {
+            for idx in posting_list {
+                if self.current_offset == self.backing_file_size {
+                    self.new_backing_file()?;
+                }
+                buffer.extend_from_slice(idx.to_le_bytes().as_ref());
+                // Write as much as possible to the current backing file
+                let available_space = self.backing_file_size - self.current_offset;
+                if buffer.len() == available_space {
+                    self.write_to_current_mmap(&buffer)?;
+                    buffer.clear();
+                }
+            }
+        }
+        // Write any remaining data in the buffer after processing all posting lists
+        if !buffer.is_empty() {
+            self.write_to_current_mmap(&buffer)?;
+            buffer.clear();
+        }
+        self.resident = false;
+        self.resident_posting_lists.clear();
+        Ok(())
+    }
+
+    // The caller is responsible for setting self.current_offset to the right value
+    fn append_posting_list_to_disk(&mut self, posting_list: &[u64]) -> Result<()> {
+        if self.resident {
+            return Err(anyhow!("Posting lists should already be flushed to disk"));
+        }
+        let usize_in_bytes = std::mem::size_of::<u64>();
```

Comment:
move this to line 196. also the name is not very clear. maybe `id_size_in_bytes` or something